ambry-network/src/main/java/com/github/ambry/network/SocketRequestResponseChannel.java

```diff
@@ -145,7 +162,10 @@ public SocketRequestResponseChannel(int numProcessors, int queueSize) {
   /** Send a request to be handled, potentially blocking until there is room in the queue for the request */
   @Override
   public void sendRequest(NetworkRequest request) throws InterruptedException {
-    requestQueue.put(request);
+    if (!networkRequestQueue.offer(request)) {
```

Comment:
Same here. 
If NetworkRequestQueue hold all the requests, we don't need unqueuedRequest queue for each RequestResponseChannel implementation.